sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java

```diff
@@ -624,7 +633,9 @@ public void onConfigurationUpdated(String appSecret, String transmissionTargetTo
      */
     @WorkerThread
     private void setDefaultTransmissionTarget(String transmissionTargetToken) {
-        mDefaultTransmissionTarget = getInstanceTransmissionTarget(transmissionTargetToken);
+        if (transmissionTargetToken != null) {
+            mDefaultTransmissionTarget = getInstanceTransmissionTarget(transmissionTargetToken);
```

Comment:
This is already covered in unit test. Without the if, mDefaultTransmissionTarget would still be null but with an error log saying that you need to pass non null. Since we do it internally its confusing for users (this happens when starting with no secret from an app).